# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/PHttpBackgroundChannel.ipdl
# Commit: 3b268ac50692
# Full Hash: 3b268ac50692bae6b1bcd0c41e61e267a6735e68
# Author: Junior Hsu <juhsu@mozilla.com>
# Date: 2020-06-24 09:31:07
# Regressor Bug: 1633935
# File Overlap Count: 1
# Description:
#   Bug 1633935 - P9 send OnAfterLastPart via pBg IPC to avoid race between OS*R, r=mayhemer,necko-reviewers
#   
#   Depends on D77750
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D77751
# ==============================================================================

diff -r ac9c5c1e5162 -r 3b268ac50692 netwerk/protocol/http/PHttpBackgroundChannel.ipdl
--- a/netwerk/protocol/http/PHttpBackgroundChannel.ipdl	Tue Jun 23 16:58:29 2020 +0000
+++ b/netwerk/protocol/http/PHttpBackgroundChannel.ipdl	Tue Jun 23 16:58:33 2020 +0000
@@ -45,6 +45,11 @@
 
   async OnStatus(nsresult status);
 
+  // Forwards nsIMultiPartChannelListener::onAfterLastPart. Make sure we've
+  // disconnected our listeners, since we might not have on the last
+  // OnStopRequest.
+  async OnAfterLastPart(nsresult aStatus);
+
   // Parent has been suspended for diversion; no more events to be enqueued.
   async FlushedForDiversion();
 